src/main/kotlin/no/nav/familie/ef/sak/iverksett/oppgaveterminbarn/ForberedOppgaverTerminbarnService.kt

```diff
@@ -0,0 +1,88 @@
+package no.nav.familie.ef.sak.iverksett.oppgaveterminbarn
+
+import no.nav.familie.ef.sak.fagsak.FagsakService
+import no.nav.familie.ef.sak.iverksett.IverksettClient
+import no.nav.familie.ef.sak.opplysninger.personopplysninger.PersonService
+import no.nav.familie.ef.sak.opplysninger.personopplysninger.domene.BarnMedIdent
+import no.nav.familie.ef.sak.opplysninger.personopplysninger.mapper.GrunnlagsdataMapper
+import no.nav.familie.ef.sak.opplysninger.personopplysninger.pdl.gjeldende
+import no.nav.familie.kontrakter.ef.iverksett.OppgaveForBarn
+import no.nav.familie.kontrakter.ef.iverksett.OppgaverForBarnDto
+import no.nav.familie.kontrakter.ef.søknad.Fødselsnummer
+import no.nav.familie.kontrakter.felles.ef.StønadType
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDate
+import java.util.UUID
+
+@Service
+class ForberedOppgaverTerminbarnService(private val personService: PersonService,
+                                        private val fagsakService: FagsakService,
+                                        private val terminbarnRepository: TerminbarnRepository,
+                                        private val iverksettClient: IverksettClient) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    @Transactional
+    fun forberedOppgaverForUfødteTerminbarn() {
+
+        val gjeldendeBarn: Map<UUID, List<TerminbarnTilUtplukkForOppgave>> = terminbarnRepository
+                .finnBarnAvGjeldendeIverksatteBehandlingerUtgåtteTerminbarn(StønadType.OVERGANGSSTØNAD)
+                .groupBy { it.behandlingId }
+
+        logger.info("Fant totalt ${gjeldendeBarn.size} terminbarn")
+
+        gjeldendeBarn.keys.forEach {
+            val terminbarnPåSøknad = gjeldendeBarn[it] ?: emptyList()
+            if (terminbarnPåSøknad.isNotEmpty()) {
+                val fødselsnummerSøker = fagsakService.hentAktivIdent(terminbarnPåSøknad.first().fagsakId)
+                val pdlBarnUnder18år = pdlBarnUnder18år(fødselsnummerSøker)
+                val ugyldigeTerminbarn = terminbarnPåSøknad.filter { !it.match(pdlBarnUnder18år) }
+                val oppgaver = lagreOgLagOppgaverForUgyldigeTerminbarn(ugyldigeTerminbarn, fødselsnummerSøker)
+                if (oppgaver.isNotEmpty()) {
+                    sendOppgaverTilIverksett(oppgaver)
+                }
+            }
+        }
```

Comment:
Hvis man her i stedet iterererte over values, så trenger man ikke `val terminbarnPåSøknad = gjeldendeBarn[it] ?: emptyList()` og heller ikke en `isNotEmpty`
```suggestion
        gjeldendeBarn.values.forEach { terminbarnPåSøknad ->
            val fødselsnummerSøker = fagsakService.hentAktivIdent(terminbarnPåSøknad.first().fagsakId)
            val pdlBarnUnder18år = pdlBarnUnder18år(fødselsnummerSøker)
            val ugyldigeTerminbarn = terminbarnPåSøknad.filter { !it.match(pdlBarnUnder18år) }
            val oppgaver = lagreOgLagOppgaverForUgyldigeTerminbarn(ugyldigeTerminbarn, fødselsnummerSøker)
            if (oppgaver.isNotEmpty()) {
                sendOppgaverTilIverksett(oppgaver)
            }
        }
```

Reply:
Bra! Endrer